synapse/handlers/account_validity.py

```diff
@@ -37,29 +37,38 @@ def __init__(self, hs):
         self.sendmail = self.hs.get_sendmail()
         self.clock = self.hs.get_clock()
 
-        self._account_validity = self.hs.config.account_validity
+        self._account_validity_enabled = self.hs.config.account_validity_enabled
+        self._account_validity_renew_by_email_enabled = (
+            self.hs.config.account_validity_renew_by_email_enabled
+        )
         self._show_users_in_user_directory = self.hs.config.show_users_in_user_directory
         self.profile_handler = self.hs.get_profile_handler()
 
+        if self._account_validity_enabled:
+            self._account_validity_period = self.hs.config.account_validity_period
```

Comment:
It might be clearer to set `account_validity_period` to `None` if account validity is not enabled in the config file, but up to you if you want to make the change.

Reply:
Yeah, there's actually a few instances of this, and doing so would probably be clearer.